ros2bag/ros2bag/verb/play.py

```diff
@@ -34,22 +35,14 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
                  'playback. Larger size will result in larger memory needs but might prevent '
                  'delay of message playback.')
         parser.add_argument(
-            '-r', '--rate', type=self.check_positive_float, default=1.0,
+            '-r', '--rate', type=check_positive_float, default=1.0,
             help='rate at which to play back messages. Valid range > 0.0.')
 
-    def check_positive_float(self, value):
-        try:
-            fvalue = float(value)
-            if fvalue <= 0.0:
-                raise ArgumentTypeError('%s is not in the valid range (> 0.0)' % value)
-            return fvalue
-        except ValueError:
-            raise ArgumentTypeError('%s is not of the valid type (float)' % value)
-
     def main(self, *, args):  # noqa: D102
         bag_file = args.bag_file
         if not os.path.exists(bag_file):
```

Comment:
Why not reusing argparse.FileType instead of reimplementing the logic?

Reply:
Good point